include/fastcdr/Cdr.h

```diff
@@ -1453,7 +1459,7 @@ namespace eprosima
                     Cdr& serializeArray(const std::string *string_t, size_t numElements)
                     {
                         for(size_t count = 0; count < numElements; ++count)
-                            serialize(string_t[count].c_str());
+                            serialize(string_t[count]);
```

Comment:
> According to CDR specification: The string contents include a single terminating null character. The string length includes the null character, so an empty string has a length of 1.

That is the point of this pull request. I think C++ `std::string` is not what CDR calls "string". It's an implementation detail, but a `std::string` seems to match a CDR sequence more than a CDR string. The CDR string seems more like a C string.

> With this change, the final null character is not sent. It is possible Fast-RTPS loses interoperability with this change.

This would require a recompile on all sides. Also, it assumes that users won't mix `std::string` with C strings.

Reply:
According to IDL mapping to C++: `The OMG IDL unbounded string type is mapped to std::string. `. The main purpose of `fastcdr` is to be a serialization library for C++ code generated from an OMG IDL. `std::string` should be serialized as a IDL/CDR string.

Reply:
I see. Thanks for the clarification. I can call `SerializeSequence` directly outside of the library for binary strings.